test/models/shared.ts

```diff
@@ -10,4 +10,6 @@ export async function gotoIfNotCurrent(
   if (pathname !== expectedPath) {
     await page.goto(expectedPath, opts);
   }
+
+  await page.waitForURL((url) => url.pathname === expectedPath);
```

Comment:
Would it work to replace all instances of `page.waitForNavigation` with this `page.waitForURL`?
I think that's what Playwright is recommending now anyway.

Reply:
Yes, I already did a batch in https://github.com/mui/mui-toolpad/pull/1699. Thought I had all of them.
But let's keep that for a follow up indeed